monai/data/dataset.py

```diff
@@ -789,8 +794,10 @@ def __init__(
         if self.num_workers is not None:
             self.num_workers = max(int(self.num_workers), 1)
         self.cache_num = 0
-        self._cache: Union[List, Dict] = []
+        self._cache: Union[List, Dict, DictProxy, ListProxy] = []
+        self.runtime_cache = runtime_cache
         self.set_data(data)
+        self.set_multiprocessing_cache(runtime_cache)
```

Comment:
perhaps `set_data` should run after `set_multiprocessing_cache`?
```suggestion
        self.set_multiprocessing_cache(runtime_cache)
        self.set_data(data)
```

Reply:
updated in the latest commit.

Thanks.

Reply:
Hi @wyli ,

Actually, here we should call `set_data()` first, the logic is to convert the set data to multiprocessing list or regular list.

Thanks.